hydroflow_lang/src/graph/ops/split.rs

```diff
@@ -0,0 +1,47 @@
+use super::{
+    OperatorConstraints, OperatorWriteOutput, WriteContextArgs, WriteIteratorArgs, RANGE_1,
+};
+
+use quote::quote_spanned;
+use syn::parse_quote;
+
+/// > 1 input stream of pair tuples `(A, B)`, 2 output streams
```

Comment:
Any performance hit for writing `map(|x| lambda(x) -> split()` instead of supporting `split(|x| lambda(x))`?

Reply:
Beyond that, do we prefer the idea of split with a built-in splitter closure? Maybe we need to talk separately and holistically about the structure-assuming operators: `join`, `groupby`, `split`... I spend a fair bit of my hydroflow debugging time fixing `join` inputs and outputs. I'll add an issue on that.